Yank/yank.py

```diff
@@ -411,6 +411,14 @@ def number_of_iterations(self):
     def number_of_iterations(self, value):
         self._sampler.number_of_iterations = value
 
+    @property
+    def is_complete(self):
+        """bool: is the sampler complete by some other mechanism. If no method is present in sampler, returns False"""
+        try:
+            return self._sampler.is_complete
+        except AttributeError:
+            return False
```

Comment:
Alternatively, you could check `self.iteration == self.number_of_iterations` if sampler doesn't expose a `is_complete` attribute.